server/metrics.go

```diff
@@ -0,0 +1,41 @@
+//go:build !no_metrics
+// +build !no_metrics
+
+package server
+
+import (
+	"net/http"
+
+	"github.com/docker/go-metrics"
+	"github.com/gorilla/mux"
+)
+
+// namespacePrefix is the namespace prefix used for prometheus metrics.
+const namespacePrefix = "notary_server"
+
+// Server uses handlers.Changefeed for two separate routes. It's not allowed
+// to register twice ("duplicate metrics collector registration attempted"),
+// so checking if it's already instrumented, otherwise skip.
+var instrumented = map[string]struct{}{}
+
+// instrumentedHandler instruments a server handler for monitoring with prometheus.
+func instrumentedHandler(handlerName string, handler http.Handler) http.Handler {
+	if _, registered := instrumented[handlerName]; registered {
+		// handler for this operation is already registered.
+		return handler
+	}
```

Comment:
Had to add this to prevent the `handlers.Changefeed` being registered twice; it's used for two endpoints, bot using `Changefeed` as handlerName / operationName. Double check if it still works as expected with this.